packages/core/src/signals/signal-provider.ts

```
@@ -2,6 +2,7 @@ import type { Agent } from '../agent/agent';
 import type { Mastra } from '../mastra';
 import type { SendNotificationSignalInput } from '../notifications/types';
 import type { InputProcessorOrWorkflow, OutputProcessorOrWorkflow } from '../processors';
+import type { AgentSignalIfIdleOptions } from '../agent/types';
```

ESLint

[error] `../agent/types` type import should occur before type import of `../mastra`

(import/order)